src/cmd/flux-exec.c

```diff
@@ -394,6 +428,20 @@ int main (int argc, char *argv[])
     if (flux_get_size (h, &rank_range) < 0)
         log_err_exit ("flux_get_size");
 
+    if (optparse_hasopt (opts, "with-imp")) {
+        if (!(imp_path = flux_imp_path (h)))
+            log_err_exit ("--with-imp: exec.imp path not found in config");
+        use_imp = true;
+        if (flux_cmd_argv_insert (cmd, 0, "run") < 0
+            || flux_cmd_argv_insert (cmd, 0, imp_path))
```

Comment:
Second `flux_cmd_argv_insert()` should check for < 0 like the first.

Reply:
D'oh!